### ISSUE TITLE
net.Server.listen's backlog option is no effect when use pm2

### ISSUE THREAD

**Issue by tangxinfa** [Unitech/pm2#1786]
My server's backlog set to 4096 when i call net.Server.listen

When start with pm2, backlog is 511(default value):

```
# ss -lt | grep 5170
LISTEN     0      511                     *:5170                     *:*       
```

If  use node, backlog is 4096:

```
# ss -lt | grep 5170
LISTEN     0      4096                    *:5170                     *:*       
```

My server's node version: v0.12.7, pm2 version: 0.14.3

**Comment by tangxinfa** [comment]
This problem looks like come from node.js 's cluster module, still don't know how to resolve it.

**Comment by Unitech** [comment]
What is that backlog?

**Comment by tangxinfa** [comment]
There is a two example, i set backlog to 64:

## Without cluster:

    var http       = require('http');
    
    http.createServer(function(req, res) {
        res.writeHead(200);
        res.end("hello world\n");
    }).listen(3000, "0.0.0.0", 64);

ss -ltn | grep 3000 output:

    LISTEN     0      64           *:3000                     *:*                  

## With cluster

    var cluster    = require('cluster'),
        http       = require('http');
    
    
    if (cluster.isMaster) {
        cluster.fork();
    } else {
        http.createServer(function(req, res) {
            res.writeHead(200);
            res.end("hello world\n");
        }).listen(3000, "0.0.0.0", 64);
    }

ss -ltn | grep 3000 output:

    LISTEN     0      511          *:3000                     *:*                  

The backlog take effect is 511 not 64, this means node cluster module NOT take care of the backlog option when we call listen: [server.listen(port[, hostname][, backlog][, callback])](https://nodejs.org/docs/latest-v0.12.x/api/http.html#http_server_listen_port_hostname_backlog_callback)

**Comment by tangxinfa** [comment]
Already pulled a request to [node](https://github.com/nodejs/node): [ cluster, net: backlog need pass to listen #4056 ](https://github.com/nodejs/node/pull/4056)

**Comment by eladnava** [comment]
Another PR (more recent), which is also now closed: https://github.com/nodejs/node/pull/33827

**Comment by eladnava** [comment]
This is now fixed in PR https://github.com/nodejs/node/pull/41623 and published in Node `17.6.0`!
